app/controllers/mass_enrollment_controller.rb

```diff
@@ -13,29 +13,7 @@ def index
 
   def add_users
     @course = Course.find_by(slug: params[:course_id])
-
-    usernames_list = params[:usernames].lines.map(&:strip)
-    @results = {}
-    usernames_list.each do |username|
-      @results[username] = add_user(username)
-    end
-
-    UpdateCourseWorker.schedule_edits(course: @course, editing_user: current_user)
-
+    AddUsers.new(course: @course, usernames: params[:usernames], current_user: current_user).add_all_at_once
```

Comment:
To keep the results behavior we currently have, you need to make `add_all_at_once` return the `results`, and assign that to `@results` here. (I guess that's what you were talking about in issue #1812?)

`@results` gets used by the corresponding view for this controller, to render the success/failure of each username submitted.